### PR TITLE
React Revamp: Props lesson

### PR DESCRIPTION
**Pull request by bscottnz**
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**
Closes https://github.com/TheOdinProject/top-meta/issues/229

**2. This PR:**
Adds props lesson to react rewrite

**3. Additional Information:**

### CODE REVIEW

File: react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+Button.defaultProps = {
+  text: "Click Me!",
+  color: "blue",
+  fontSize: 12
+};
+
+export default function App() {
+  return (
+    <div>
```
Comment (01zulfi) [inline react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md:143]: How would you feel about adding in this section that default props can be set in the component definition itself?

```jsx
function Button({ text, color, fontSize = 12 }) {
...
```

I'll let you decide if we want to add this information or not.

File: react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+1.  Read through the React Docs Beta on [Passing Props to a Component](https://beta.reactjs.org/learn/passing-props-to-a-component). Make sure to edit the code examples and experiment with different prop values.
+</div>
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
+
+*   <a class="knowledge-check-link" href="#data-transfer-in-react">How does data flow between React components? From child to parent? From parent to child? Both?</a>
+*   <a class="knowledge-check-link" href="#using-props-in-react">Why do we use props in React?</a>
+*   <a class="knowledge-check-link" href="A KNOWLEDGE CHECK URL">How do we define default properties on a React component? What are some benefits in doing so?</a>
```
Comment (01zulfi) [inline react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md:168]: 404 link

**Review by 01zulfi — CHANGES_REQUESTED** [review]
Solid and concise lesson you have here 🔥 🔥 

Left some comments below.

An addition I think we can make is that functions can be passed in as props, and then can be invoked within the component. Thoughts?

File: react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md (hunk 1 above)
Comment (bscottnz) [reply react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md:143]: TIL. I've added a brief mention.

File: react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md (hunk 2 above)
Comment (bscottnz) [reply react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md:168]: fixed

**Comment by bscottnz** [comment]
In regards to showcasing functions as props. I was thinking about this, but wasn't sure where this lesson fits in with respect to React events. Looking through https://github.com/TheOdinProject/top-meta/issues/194 I don't see any mention of events, but it may be covered in one of the previous introductory lessons? I could add something basic such as a function that links to the supplied URL, but unsure how to approach onClick etc if it hasn't been covered. Thoughts?

File: react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+  return (
+    <div>
+      <Button text="Click Me!" color="blue" fontSize={12} />
+      <Button text="Don't Click Me!" color="red" fontSize={12} />
+      <Button text="Click Me!" color="blue" fontSize={20} />
+    </div>
+  );
+}
+~~~
+### Default Props
```
Comment (01zulfi) [inline react/getting_started_with_react/passing_data_between_components/passing_data_between_components.md:121]: Can we add a linebreak before the Default Props heading? Our markdown parser wants a linebreak here🤷 

image
```suggestion
~~~

### Default Props
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]
> In regards to showcasing functions as props. I was thinking about this, but wasn't sure where this lesson fits in with respect to React events. Looking through https://github.com/TheOdinProject/top-meta/issues/194 I don't see any mention of events, but it may be covered in one of the previous introductory lessons? I could add something basic such as a function that links to the supplied URL, but unsure how to approach onClick etc if it hasn't been covered. Thoughts?


I don't think event handling is explicitly covered in the previous lessons; I'm not too worried about it as I believe learners can pick it up easily as they progress through these lessons. And anyway, the next 'States and Effects' section will inevitably include examples covering events.

I say, go ahead with the `onClick` example as it's on point with the Button component we've been building throughout this lesson.

**Comment by bscottnz** [comment]
Functions as props have been added.

**Review by 01zulfi — APPROVED** [review]

**Pull request merged**